plugins/sparc2_alignment_data_collector.py

```diff
@@ -0,0 +1,699 @@
+                            self._resume_requested.clear()
+                            self._resume_requested.wait()
+                            self._aligned_pos = {
+                                "l": self.mirror.position.value["l"],
+                                "s": self.mirror.position.value["s"],
+                                "x": self.mirror_xy.position.value["x"],
+                                "y": self.mirror_xy.position.value["y"],
+                                "z": self.stage.position.value["z"],
+                            }
+                            wx.CallAfter(self.status_lbl.SetLabel, "Status: Resumed")
+                            logging.debug(
+                                "Resumed. New aligned position: %s", self._aligned_pos
+                            )
+                        self.stage.moveAbs({"z": z}).result()
+                        dl = self._aligned_pos["l"] - self.mirror.position.value["l"]
+                        ds = self._aligned_pos["s"] - self.mirror.position.value["s"]
+                        dx = self._aligned_pos["x"] - self.mirror_xy.position.value["x"]
+                        dy = self._aligned_pos["y"] - self.mirror_xy.position.value["y"]
+                        dz = self._aligned_pos["z"] - self.stage.position.value["z"]
+                        forcemd = {}
+                        forcemd[model.MD_EXTRA_SETTINGS] = {
+                            "l": self.mirror.position.value["l"],
+                            "s": self.mirror.position.value["s"],
+                            "x": self.mirror_xy.position.value["x"],
+                            "y": self.mirror_xy.position.value["y"],
+                            "z": self.stage.position.value["z"],
+                            "dl": dl,
+                            "ds": ds,
+                            "dx": dx,
+                            "dy": dy,
+                            "dz": dz,
+                            "l_aligned": self._aligned_pos["l"],
+                            "s_aligned": self._aligned_pos["s"],
+                            "x_aligned": self._aligned_pos["x"],
+                            "y_aligned": self._aligned_pos["y"],
+                            "z_aligned": self._aligned_pos["z"],
+                            "wd": self.ebeam_focus.position.value["z"],
+                        }
+                        logging.debug(f"Acquiring l={l} s={s} z={z}")
+                        filepath = os.path.join(
+                            path, f"{idx}_snapshot_{l}_{s}_{z}" + hdf5.EXTENSIONS[0]
+                        )
+                        self.tab._ccd_spe.pauseStream()
+                        self.tab._ccd_spe.pause()
+                        data = self.tab._ccd_stream.getSingleFrame()
+                        data.metadata.update(forcemd)
+                        self.tab._ccd_stream._onNewData(self.tab._ccd_stream._dataflow, data)
```

Comment:
This should work. 
That said, it'd probably more correct to use `single_frame_acquisition.value = True` and use `is_active.value = True` to trigger one frame acquisition. That's the "standard" way to use it.